packages/hawtio/src/ui/page/HawtioHeader.tsx

```diff
@@ -109,20 +109,31 @@ const HawtioHeaderToolbar: React.FunctionComponent = () => {
    * Determine which plugin is currently displaying
    * based on the path of the current location
    */
-  const pluginFromLocation = (): Plugin | null => {
+  const filterPlugins = (): Plugin[] => {
     const path = location.pathname
-    return plugins.find(plugin => path.startsWith(plugin.path)) ?? null
+    return plugins.filter(plugin => {
+      if (path.startsWith(plugin.path)) return true
+
+      if (!plugin.headerItems) return false
+
+      return plugin.headerItems.universal === true
+    })
   }
 
-  const plugin = pluginFromLocation()
+  const filteredPlugins = filterPlugins()
```

Comment:
I think we don't need to filter plugins here. Instead we can just directly collect only relevant header item components againt all the plugins here, like:
```
const headerItems: React.ComponentType<any>[] = collectHeaderItems()
```